### PR TITLE
Dependencies/CI: updates to prepare for jQuery 4.0.0-beta testing

### PR DESCRIPTION
**Pull request by jayaddison**
This pull request is dependency maintenance.

The following changes were made

- Migrate from the deprecated `node-sass` to `sass` (Dart SASS) library.
  - The CSS `outputStyle` setting is updated accordingly.
- Upgrade to `grunt-contrib-qunit` v10.1.1.
  - This requires a major version update for QUnit; here the latest v2.22.0 is selected.
- Upgrade GitHub Actions workflows to NodeJS v20 `iron` (the current Long-Term-Support release).

This changeset prepares for an upgrade to jQuery 4.0.0-beta2, and extracts unrelated dependency changes from PR #6332.

### CODE REVIEW

**Comment by jayaddison** [comment]
cc @kevin-brown

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,24 +20,29 @@ jobs:
...
         uses: actions/setup-node@v1
         with:
-          node-version: 12
+          node-version: 20
       - name: npm install
         run: npm install
       - name: Run tests
+        env:
+          PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1
+          PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser
```
Comment (kevin-brown) [inline .github/workflows/main.yml]: Getting an error on here about there being invalid syntax.
```suggestion
          PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: "1"
          PUPPETEER_EXECUTABLE_PATH: "/usr/bin/chromium-browser"
```

File: package.json [hunk 2]
```diff
@@ -53,12 +53,15 @@
...
     "grunt-contrib-requirejs": "^1.0.0",
     "grunt-contrib-uglify": "~4.0.1",
     "grunt-contrib-watch": "~1.1.0",
     "grunt-sass": "^3.1.0",
     "jquery-mousewheel": "~3.1.13",
-    "node-sass": "^7.0.3"
+    "sass": "^1.77.8"
+  },
+  "dependencies": {
+    "sass": "^1.77.8"
```
Comment (kevin-brown) [inline package.json]: Fairly certain this can be removed since this is defined as a dev dependency.
```suggestion
```

**Review by kevin-brown — CHANGES_REQUESTED** [review]

**Comment by kevin-brown** [comment]
I just want to take a moment to say thank you so much for making this pull request. Please ping me when the builds are back to running (looks like a syntax error is killing them before they start) and I'll merge this in.

File: package.json (hunk 2 above)
Comment (jayaddison) [reply package.json]: Indeed that can, thanks; this seems to have been a mistake during merge/rebase.

**Comment by jayaddison** [comment]
You're welcome; thanks @kevin-brown.  These changes should be ready now.

**Review by kevin-brown — APPROVED** [review]

**Comment by jayaddison** [comment]
@kevin-brown would you consider releasing a beta/rc of `select2` with `jquery` 4.0.0-beta support?  If so, that'd help me to upgrade both of them, in order to offer some practical testing and/or feedback.

(yep, I could pin specific `git` commit hashes in `package.json`.. but I'd prefer/find it easier to use packaged versions distributed through the usual channels if possible)

**Comment by jayaddison** [comment]
@kevin-brown NB: `jquery` v4.0.0 (final) has been released: https://github.com/jquery/jquery/releases/tag/4.0.0

**Pull request merged**